tools/osbuild-mpp

```diff
@@ -1568,6 +1609,47 @@ class ManifestFileV2(ManifestFile):
             inputs_manifests["origin"] = "org.osbuild.source"
             inputs_manifests["references"] = manifest_lists
 
+    def _process_ostree_commits(self, stage):
+        if stage.get("type", "") not in \
+            ["org.osbuild.ostree.pull", "org.osbuild.ostree.deploy"]:
+            return
+
+        inputs = element_enter(stage, "inputs", {})
+        inputs_commits = element_enter(inputs, "commits", {})
+
+        if inputs_commits.get("type", "") != "org.osbuild.ostree":
+            return
+        if inputs_commits.get("origin", "") != "org.osbuild.source":
+            return
+
+        mpp = self.get_mpp_node(inputs_commits, "resolve-ostree-commits")
+        if not mpp:
+            return
+
+        refs = element_enter(inputs_commits, "references", {})
+
+        ostree_commit_source = element_enter(self.sources, "org.osbuild.ostree", {})
+        items = element_enter(ostree_commit_source, "items", {})
+
+        for commit in element_enter(mpp, "commits", []):
+            remote = commit["remote"]
+            ref = commit["ref"]
+
+            with tempfile.TemporaryDirectory() as tmprepo:
+                ostree.cli("init", mode="archive", repo=tmprepo)
+                # This is a temporary repo so we'll just use a random name
+                remote_name = "tmpremote"
+                ostree.setup_remote(tmprepo, remote_name, remote)
+                ostree.cli( "pull", "--commit-metadata-only",
+                            "--mirror", remote_name, ref, repo=tmprepo)
+                checksum = ostree.rev_parse(tmprepo, ref)
+                items[checksum] = {
+                    "remote": remote
+                }
+                refs[checksum] = {
+                    "ref": ref
+                }
```

Comment:
I thought about doing the same - but the [subscription manager stuff](https://github.com/osbuild/osbuild/blob/f0982873e27e0cbbcca77927347ac491caf933b5/sources/org.osbuild.ostree#L117-L120) made me decide to just re-use the code that already existed rather than writing new code.